src/test/java/com/elasticbox/jenkins/tests/PullRequestManagerTest.java

```diff
@@ -0,0 +1,255 @@
+        Assert.assertTrue("Deployed instances are not deleted after 5 minutes since the pull request is closed", instances.isEmpty());
+
+        LOGGER.fine("End of test testAddingProjectToGHRepository\n");
+
+    }
+
+
+    public void testSamePullRequestsTriggersMultipleProjects(FreeStyleProject project2) throws Exception {
+        final List<JSONObject> instances = new ArrayList<JSONObject>();
+
+        LOGGER.fine("test ---- testSamePullRequestsTriggersMultipleProjects\n");
+        LOGGER.fine("Launching one pull request that triggers two jobs");
+        pullRequest.open();
+        Thread.sleep(10000);
+
+        LOGGER.fine("Check that the job 1 build is triggered");
+        waitForBuildTriggered(project, NEXTBUILD_TIMEOUT_LONG, "testSamePullRequestsTriggersMultipleProjects 1");
+
+        Assert.assertNotNull(MessageFormat.format("Build is not triggered on opening of pull request {0} after {1} minutes",
+                pullRequest.getGHPullRequest().getHtmlUrl(), TimeUnit.SECONDS.toMinutes(NEXTBUILD_TIMEOUT_LONG)),
+                project.getLastBuild() );
+
+
+        LOGGER.fine("Check that the job 2 build is triggered");
+        waitForBuildTriggered(project2, NEXTBUILD_TIMEOUT, "testSamePullRequestsTriggersMultipleProjects 2");
+
+        Assert.assertNotNull(MessageFormat.format("Build is not triggered on opening of pull request {0} after {1} minutes",
+                pullRequest.getGHPullRequest().getHtmlUrl(), TimeUnit.SECONDS.toMinutes(NEXTBUILD_TIMEOUT_LONG + NEXTBUILD_TIMEOUT)),
+                project2.getLastBuild());
+
+
+        // Tests success. Finishing...
+        LOGGER.fine("Test completed succesfully. Cleaning...");
+
+        LOGGER.fine("Waiting for completion of buildings...");
+        waitForCompletion(project, TimeUnit.MINUTES.toSeconds(3));
+        Assert.assertFalse(MessageFormat.format("Job 1 Build of pull request {0} is still not complete after 3 minutes", pullRequest.getGHPullRequest().getHtmlUrl()),
+                project.getLastBuild().isBuilding());
+
+        waitForCompletion(project2, TimeUnit.MINUTES.toSeconds(3));
+        Assert.assertFalse(MessageFormat.format("Job 2 Build of pull request {0} is still not complete after 3 minutes", pullRequest.getGHPullRequest().getHtmlUrl()),
+                project2.getLastBuild().isBuilding());
+
+        LOGGER.fine("Closing the pull request 1");
+        instances.addAll(checkBuild(project, null));
+
+        LOGGER.fine("Closing the pull request and deleting " + instances.size() + " instances.");
+        pullRequest.close();
+        waitForDeletion(instances, TimeUnit.MINUTES.toSeconds(5));
+        Assert.assertTrue("Deployed instances are not deleted after 5 minutes since the pull request is closed", instances.isEmpty());
+
+        LOGGER.fine("End of testSamePullRequestsTriggersMultipleProjects\n");
+
+
+    }
+
+    private FreeStyleProject createJenkinsJob2 () throws Exception {
```

Comment:
Maybe it's best to have (maybe even it already exists) a method to pass a name and a file, and create the Project for it

Reply:
I concur, also the name is not clear, it could be that there is another method createJenkinsJob or that the method creates the job 2.

Reply:
Yes, so I did. As you can see the method initializes in first sentences two variables that can be considered as parameters if necessary. But later I saw that a second project only is required in this particular test class. That's why for clarity I preferred to leave it differentiated as (project2) although prepared for an easy parametrization.